published/external/xdpapi_helper.h

```diff
@@ -0,0 +1,229 @@
+
+#pragma once
+
+#include "xdpapi.h"
+
+#if defined(_KERNEL_MODE)
+
+extern XDP_API_PROVIDER_DISPATCH *XdpApiTest;
+extern XDP_API_PROVIDER_BINDING_CONTEXT *ProviderBindingContext;
+
+#define XDP_CALL_WITH_RETURN(ReturnType, FunctionCall, ...) \
+    do { \
+        ((XDP_API_CLIENT *)(ProviderBindingContext))->InProgressCallCount++; \
```

Comment:
this needs to be thread-safe, and ideally also scalable. i think an `EX_RUNDOWN_REF_CACHE_AWARE` would be perfect, because it can also prevent new calls when NMR is trying to unbind